gen_funcs.sh

```diff
@@ -1853,20 +1853,21 @@ check_disk_space_requirements() {
 			gen_die "--check-free-disk-space-bootdir value '${CHECK_FREE_DISK_SPACE_BOOTDIR}' is not a valid number!"
 		fi
 
-		available_free_disk_space=$(unset POSIXLY_CORRECT && LC_ALL="C" df -BM "${BOOTDIR}" | awk '$3 ~ /[0-9]+/ { print $4 }')
+		available_free_disk_space=$(expr "$(stat -fc%f "${BOOTDIR}")" \* "$(stat -fc%S "${BOOTDIR}")")
```

Comment:
``` C
available_free_disk_space=$(($(stat -fc '%f*%S'"${BOOTDIR}")))
````

Reply:
Yeah, since we are using GNU bash as the interpreter, that works and helps make a single call to `stat(1)` insead of two.

My programming style is geared towards code portability, hence I use the POSIX shell syntax.

Since both variants are correct and the difference in script execution time is negligible, I'll defer to the maintainer on the choice of syntax.

Reply:
I usually prefer to write scripts in a portable way, but since genkernel is tied to bash, why not to use bash advantages?

BTW, `$(())` mathematics is not a bashism, it is standard POSIX:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_04

Reply:
Thanks for pointing out that `$(())` arithmetic expansion is in fact POSIX. I will update the PR.

> since genkernel is tied to bash, why not to use bash advantages?

In general, while genkernel as a whole is tied to bash, we might want to reuse parts of it in other scripts. Checking for free disk space seems like something that can be useful even outside of genkernel context.